### PR TITLE
feat: create appoinment

### PR DESCRIPTION
**Pull request by Simpsonkk**
## Describe your changes
- I did endpoints for creating appointments and getting activities, capabilities, and diagnoses lists.
- I subtract the amount for one hour from the seeker, if he has enough money and save this number, if the user does not have enough money, then I do nothing - they told me to do this for now.

- I switched off eslint rule 'class-methods-use-this'. I believe we shouldn't use it especially in nest js for some reasons:
1. NestJS most often uses services with non-static methods, for example, dynamic dependency management through dependency injection. If most of your services use non-static methods, this is the style of negotiation you should use.
2. One of the key features of NestJS is its powerful dependency injection system. Static methods cannot use injected dependencies directly because they belong to the class and not to the instance of the class. This limits their functionality and flexibility in the context of NestJS, where dependency injection is the primary way to manage dependent services and configuration.
3. Non-static methods make it easier to modify and extend the behavior of services because they can use state and instance methods. This makes the code more flexible and easier to maintain.
4. NestJS is focused on object-oriented programming. Using non-static methods better aligns with OOP principles such as encapsulation, inheritance, and polymorphism. This allows developers to organize and structure their code more efficiently.

DB schema: 
image

## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/AC-115?atlOrigin=eyJpIjoiZWFiZGQxMzMwNjJjNDFjZGIzODdkM2E5Mzk3MmM1ZmQiLCJwIjoiaiJ9)

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [x] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

File: src/common/entities/appointment.entity.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    description: 'Additional details of the appointment',
+  })
+  @Column({ nullable: true })
+  details: string;
+
+  @ApiProperty({
+    description: 'Payment for one hour',
+    example: '10',
+  })
+  @Column({ default: 0, type: 'decimal' })
```
Comment (ProggerOleg) [inline src/common/entities/appointment.entity.ts]: hourlyRate in caregiver_info table is 'int' type, lets have same type

File: src/common/types/decoded-token.type.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export type DecodedToken = {
+  id: '2dc65c36-5112-4901-96b8-2e7520ab57c8';
+  iat: 1701276974;
+  exp: 1701536174;
+};
```
Comment (ProggerOleg) [inline src/common/types/decoded-token.type.ts]: I belive you forgot to delete this file(you don't use it in this PR) and it's obviously not type)

Comment (Simpsonkk) [reply src/common/types/decoded-token.type.ts]: yep, it was like a draft, thanks, removed

File: src/common/entities/appointment.entity.ts (hunk 1 above)
Comment (Simpsonkk) [reply src/common/entities/appointment.entity.ts]: I thought it might be fractional, ok

**Review by Lilly-nn — APPROVED** [review]

**Review by fedasevich — APPROVED** [review]

**Review by ProggerOleg — APPROVED** [review]

**Pull request merged**